solr/contrib/ltr/src/java/org/apache/solr/ltr/LTRRescorer.java

```diff
@@ -131,31 +146,12 @@ public TopDocs rescore(IndexSearcher searcher, TopDocs firstPassTopDocs,
   }
 
   protected static void sortByScore(ScoreDoc[] reranked) {
```

Comment:
So now the `sortByScore` contains much less code than before. I wonder if its future might look like this:
* step 1: mark it deprecated (in this pull request) so that no new code starts calling it
* step 2: change existing code (in this pull request) to sort directly with the new comparator (which already has protected visibility equivalent to the visibility of this method)
* step 3: remove the deprecated method (after this pull request is merged) on `main` branch only for Solr 9.0 onwards but retain it in `branch_8x` branch for Solr 8.10 onwards, to maintain build backwards compatibility for anyone who may have built a custom plugin with a custom rescorer.

```suggestion
  @Deprecated
  protected static void sortByScore(ScoreDoc[] reranked) {
```

Reply:
Ok, Christine! I followed your instructions and made step1 and step2 modifications. About the step3, do I need to open a new Issue for that?

Reply:
> ... About the step3, do I need to open a new Issue for that?

Good question. I would take care of that after merging and backporting of the pull request, no need for a separate issue or pull request, thanks for checking.

Reply:
Nice :-)